### PR TITLE
Bump core library to 3.7.1, add support for cellArea, exactEdgeLength, pointDist

### PR DESCRIPTION
**Pull request by nrabinowitz**
- Bumps core H3 library to 3.7.1
- Adds support and tests for `cellArea`, `exactEdgeLength`, and `pointDist` functions

### CODE REVIEW

File: lib/h3core.js [hunk 1]
```diff
@@ -69,7 +69,9 @@ export const UNITS = {
     m: 'm',
     km: 'km',
     m2: 'm2',
-    km2: 'km2'
+    km2: 'km2',
```
Comment (isaacbrodsky) [inline lib/h3core.js:72]: nit: this is an odd grouping of types, seems like km/km2 should be next to each other, or all the area ones should be grouped together

File: lib/h3core.js [hunk 2]
```diff
@@ -1094,7 +1109,86 @@ export function experimentalLocalIjToH3(origin, coords) {
...
-// Public informational utilities
+// Public API functions: Distance/area utilities
+
+/**
+ * Great circle distance between two geo points. This is not specific to H3,
+ * but is implemented in the library and provided here as a convenience.
+ * @static
+ * @param  {number[]} latlng1 Origin coordinate as [lat, lng]
+ * @param  {number[]} latlng2 Destination coordinate as [lat, lng]
+ * @param  {string}   unit    Distance unit (either UNITS.m or UNITS.km)
```
Comment (isaacbrodsky) [inline lib/h3core.js:1120]: or rads

File: package.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
   "scripts": {
     "build-update-h3": "bash scripts/update-h3.sh",
     "build-emscripten": "yarn build-update-h3 && yarn docker-emscripten",
-    "build-docs": "jsdoc2md --global-index-format grouped --partial doc-files/scope.hbs --helper ./doc-files/insert-version.js --separators --template doc-files/README.tmpl.md lib/h3core.js > README.md",
+    "build-docs": "jsdoc2md --no-cache --global-index-format grouped --partial doc-files/scope.hbs --helper ./doc-files/insert-version.js --separators --template doc-files/README.tmpl.md lib/h3core.js > README.md",
```
Comment (nrabinowitz) [inline package.json:31]: Pretty sure that this + the JSDoc bump should fix problems in #91 - cc @isaacbrodsky

File: test/h3core.spec.js [hunk 4]
```diff
@@ -15,7 +15,7 @@
  */
 
 import test from 'tape';
-import * as h3core from '../lib/h3core.js';
+import * as h3 from '../lib/h3core.js';
```
Comment (nrabinowitz) [inline test/h3core.spec.js:18]: This change introduced a lot of noise into this file - the only new code here are the tests for the new functions.

File: lib/h3core.js (hunk 2 above)
Comment (nrabinowitz) [reply lib/h3core.js:1120]: fixed

File: lib/h3core.js (hunk 1 above)
Comment (nrabinowitz) [reply lib/h3core.js:72]: updated

File: test/h3core.spec.js [hunk 5]
```diff
@@ -1500,34 +1462,148 @@ test('edgeLength', assert => {
...
+        assert.ok(
+            // res 0 has high distortion of average area due to high pentagon proportion
+            res === 0 ||
+                // This seems to be the lowest factor that works for other resolutions
+                almostEqual(cellAreaKm2, h3.hexArea(res, h3.UNITS.km2), cellAreaKm2 * 0.4),
+            `Area is close to average area at res ${res}`
+        );
+        // Just property tests for the other units
+        assert.ok(h3.cellArea(h3Index, h3.UNITS.m2) > cellAreaKm2, 'm2 > Km2');
+        assert.ok(cellAreaKm2 > h3.cellArea(h3Index, h3.UNITS.rads2), 'Km2 > rads2');
```
Comment (dfellis) [inline test/h3core.spec.js:1518]: Not blocking, but why not take the expected areas from the average area function for the given resolution and then assert that the ratio of the two areas is below a maximum distortion factor?

File: test/h3core.spec.js [hunk 6]
```diff
@@ -1500,34 +1462,148 @@ test('edgeLength', assert => {
...
+                    almostEqual(lengthKm, h3.edgeLength(res, h3.UNITS.km), lengthKm * 0.2),
+                `Edge length is close to average edge length at res ${res}`
+            );
+            // Just property tests for the other units
+            assert.ok(h3.exactEdgeLength(edge, h3.UNITS.m) > lengthKm, 'm > Km');
+            assert.ok(lengthKm > h3.exactEdgeLength(edge, h3.UNITS.rads), 'Km > rads');
+        }
+    }
+    assert.end();
+});
```
Comment (dfellis) [inline test/h3core.spec.js:1554]: Same nit here.

**Review by dfellis — APPROVED** [review]

File: test/h3core.spec.js (hunk 5 above)
Comment (nrabinowitz) [reply test/h3core.spec.js:1518]: I do this for `Km`, but I assume you're asking why not do it for `m` too? I can add. I was copying `h3-java` here, and was lazy :).

No averages available for rads, though.

**Review by dfellis — APPROVED** [review]

**Pull request merged**